lib/java/PlagiarismDetection/src/pd/MySQLDB.java

```diff
@@ -239,36 +243,73 @@ private void insertMappings(LinkedList<Integer> genIds,
 			ArrayList<Result> results) throws Exception {
 		String dateTime = dateFormat.format(new java.util.Date());
 		PreparedStatement stmt = con.prepareStatement(MAPPING_INSERT);
+		PreparedStatement bMappingStatement = con.prepareStatement(SKELETON_MAPPING_INSERT);
 
 		Iterator<Integer> idIterator = genIds.iterator();
 		Iterator<Result> resultIterator = results.iterator();
 
 		ArrayList<Mapping> mappings;
-		int id;
+		long mappingId = getCurrentMaxMappingId();
+		int submissionSimilarityId;;
 		while (idIterator.hasNext() || resultIterator.hasNext()) {
 
-			id = idIterator.next();
+			submissionSimilarityId = idIterator.next();
 			mappings = resultIterator.next().getCodeIndexMappings();
+			mappingId += 1;
 
 			for (Mapping m : mappings) {
-				stmt.setInt(1, id);
-				stmt.setInt(2, m.getStartIndex1());
-				stmt.setInt(3, m.getEndIndex1());
-				stmt.setInt(4, m.getStartIndex2());
-				stmt.setInt(5, m.getEndIndex2());
-				stmt.setInt(6, m.getStartLine1());
-				stmt.setInt(7, m.getEndLine1());
-				stmt.setInt(8, m.getStartLine2());
-				stmt.setInt(9, m.getEndLine2());
-				stmt.setInt(10, m.getMappedCountableStmtCount());
-				stmt.setBoolean(11, m.isPlagMapping());
-        stmt.setString(12, dateTime);
+				stmt.setLong(1, mappingId);
+				stmt.setInt(2, submissionSimilarityId);
+				stmt.setInt(3, m.getStartIndex1());
+				stmt.setInt(4, m.getEndIndex1());
+				stmt.setInt(5, m.getStartIndex2());
+				stmt.setInt(6, m.getEndIndex2());
+				stmt.setInt(7, m.getStartLine1());
+				stmt.setInt(8, m.getEndLine1());
+				stmt.setInt(9, m.getStartLine2());
+				stmt.setInt(10, m.getEndLine2());
+				stmt.setInt(11, m.getMappedCountableStmtCount());
+				stmt.setBoolean(12, m.isPlagMapping());
         stmt.setString(13, dateTime);
```

Comment:
Indent appropriately?

Reply:
I checked in VS Code, the indent is proper. I guess maybe it's a Git issue?